services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/scheduler/_core/_events_utils.py

```diff
@@ -442,3 +445,18 @@ async def prepare_services_environment(
     )
 
     scheduler_data.dynamic_sidecar.is_service_environment_ready = True
+
+
+async def get_hardware_info(
+    app: FastAPI, scheduler_data: SchedulerData
+) -> HardwareInfo | None:
+    rut_client = ResourceUsageTrackerClient.get_from_state(app)
+    try:
+        result = await rut_client.get_default_pricing_and_hardware_info(
+            product_name=scheduler_data.product_name,
+            service_key=scheduler_data.key,
+            service_version=scheduler_data.version,
+        )
+        return HardwareInfo(aws_ec2_instances=result.aws_ec2_instances)
+    except PricingPlanUnitNotFoundError:
+        return None
```

Comment:
aha, I see the relation with pricing plan now. 
Nonetheless, I still think the check you do above regarding `aws_ec2_instances` should be done here except if there is another use case in the code (which I do not see at the moment)